### PR TITLE
Fix api definition retrieval

### LINKED ISSUE #110: GET /{api_id}/versions/{version_id}/definitions/{definition_id} Entity is queried by technical id
**Current behavior:**
GET /{api_id}/versions/{version_id}/definitions/{definition_id} 
queries the table api_version via technical id

```
apiRepository.findOne(id)
```

Impact: 
Api definition <-> implementation mismatch . Client receives wrong Resource. 

**Expected behavior:**

ApiEntity should be selected by api_id and version_id and definition_id

```
apiRepository.findByApiNameAndApiVersionAndDefinitionId(..)
```

### PR DESCRIPTION
**Pull request by fmueller**
Closes #110 

This PR:
- updates dependencies of storage
- removes `id` field from definition payload since there is no valid single id (the id is the URI itself)
- fixes fetching of api definition
- merges duplicate test classes
- removes a lot of empty lines in the already touched files

### CODE REVIEW

**Comment by maxim-tschumak** [comment]
I think we should drop `api_version.id` field. It's not needed anymore.

**Comment by maschleg** [comment]
@maxim-tschumak then we have a composite foreign key of three columns in api_deployment table. I think that would just add complexity.

File: storage/src/test/java/org/zalando/apidiscovery/storage/api/ApiDefinitionResourceIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+import static org.zalando.apidiscovery.storage.DomainObjectGen.APP1_NAME;
+import static org.zalando.apidiscovery.storage.DomainObjectGen.DEFINITION;
+import static org.zalando.apidiscovery.storage.DomainObjectGen.DEFINITION_TYPE;
+import static org.zalando.apidiscovery.storage.DomainObjectGen.givenApiDeployment;
+import static org.zalando.apidiscovery.storage.DomainObjectGen.givenApiEntity;
+import static org.zalando.apidiscovery.storage.DomainObjectGen.givenApplication;
+import static org.zalando.apidiscovery.storage.TestDataHelper.readResource;
+import static org.zalando.apidiscovery.storage.api.ApiLifecycleState.ACTIVE;
+
+public class ApiDefinitionResourceIntegrationTest extends AbstractResourceIntegrationTest {
```
Comment (maschleg) [inline storage/src/test/java/org/zalando/apidiscovery/storage/api/ApiDefinitionResourceIntegrationTest.java]: Regarding our new Teststrategie this should be an component test.

Comment (fmueller) [inline storage/src/test/java/org/zalando/apidiscovery/storage/api/ApiDefinitionResourceIntegrationTest.java]: Good point, will change it.

**Comment by tkrop** [comment]
👍

**Comment by fmueller** [comment]
Do not merge yet pls

**Comment by fmueller** [comment]
@maschleg I incorporated your change requests. Please review again.

**Comment by fmueller** [comment]
FYI: codacy warnings are wrong, each test has at least one assertion.

**Comment by maschleg** [comment]
👍

**Pull request merged**